src/ucp/wireup/wireup.c

```diff
@@ -967,6 +1042,108 @@ ucp_wireup_get_reachable_mds(ucp_ep_h ep,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_check_config_intersect(ucp_ep_h ep,
+                                  ucp_ep_config_key_t *new_key,
+                                  ucs_queue_head_t *replay_pending_queue)
+{
+    ucp_worker_h worker                            = ep->worker;
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = { UCP_NULL_LANE };
+    uct_ep_h new_uct_eps[UCP_MAX_LANES]            = { NULL };
+    ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane, reuse_lane;
+    ucp_wireup_ep_t *cm_wireup_ep;
+
+    ucs_queue_head_init(replay_pending_queue);
+
+    if ((ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) ||
+        !ucp_ep_has_cm_lane(ep) ||
+        !ucp_ep_is_sockaddr_stub(ep)) {
+        return;
+    }
+
+#if UCS_ENABLE_ASSERT
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        ucs_assert(ep->uct_eps[lane] != NULL);
+    }
+#endif
+
+    old_key = &ucp_ep_config(ep)->key;
+
+    ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
+
+    if (ucp_ep_config_key_has_cm_lane(new_key)) {
+        /* CM lane has to be re-used by the new EP configuration */
+        ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+        ucs_assert(old_key->wireup_lane != UCP_NULL_LANE);
```

Comment:
is this true also if we create the first stage connection using dcx (which is 2iface transport without cb_async, so ucp_wireup_select_wireup_msg_lane() would return NULL_LANE for it) ?